File: 2ºP - Desenvolvimento Back-End/Desenvolvimento com Java/TPs/TP3/ex01/Pirata.java
package com.codingloria.TP3.ex01;

public class Pirata {
    private String nome;
    private double recompensa;

    public Pirata() {
    }

    public Pirata(String nome, double recompensa) {
        this.nome = nome;
        this.recompensa = recompensa;
    }

    @Override
    public String toString() {
        return "---- Pirata ----\nNome: " + nome + "\nRecompensa: " + recompensa + " berries\n";
    }

    public void atacar(){
        System.out.println(nome + " partiu para o ataque!");
    }

    void aumentarRecompensa(double valor){
        recompensa += valor;
        System.out.println("Recompensa de " + nome + " aumentou para " + recompensa + " berries.");
    }

    public double getRecompensa() {
        return recompensa;
    }

    public void setRecompensa(double recompensa) {
        this.recompensa = recompensa;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }
}

File: 2ºP - Desenvolvimento Back-End/Desenvolvimento com C#/TPs/TP3/ex01/Models/Pirata.cs
namespace ex01;

class Pirata {
    public String Nome { get; set; }
    public double Recompensa { get; set; }
    
    public Pirata() { }

    public Pirata(string nome, double recompensa){
        Nome = nome;
        Recompensa = recompensa;
    }
    
    public override string ToString() {
        return "--- Pirata ---\nNome: " + Nome + "\nRecompensa: " + Recompensa + " berries\n";
    }
    
    public void Atacar() {
        Console.WriteLine(Nome + " partiu para o ataque!");
    }

    public void AumentarRecompensa(double valor) {
        Recompensa += valor;
        Console.WriteLine("Recompensa de " + Nome + " aumentou para " + Recompensa + " berries.");
    }
    
}